crafters/nlp/PDFExtractor/__init__.py

```diff
@@ -14,11 +14,10 @@ def __init__(self, *args, **kwargs):
 
     def craft(self, uri: str, buffer: bytes, *args, **kwargs):
         import PyPDF2
-        import io
 
         text = ""
         if buffer:
-            pdf_obj = io.BytesIO(buffer)
+            pdf_obj = buffer
```

Comment:
buffer object has no `close`. PdfFileReader accepts a `stream` that is what `io.BytesIO` provides.